app/serializers/current_user_serializer.rb

```diff
@@ -339,7 +339,13 @@ def include_has_topic_draft?
   end
 
   def sidebar_category_ids
-    object.sidebar_categories_ids
+    categories_ids = object.category_sidebar_section_links.pluck(:linkable_id)
+
+    if categories_ids.present?
```

Comment:
object.category_sidebar_section_links.pluck(:linkable_id) will return [] if not set

Therefore, whole logic for this method can be one line like:
`object.category_sidebar_section_links.pluck(:linkable_id) & scope.allowed_category_ids`

Reply:
Good point have made the changes.